Fixes to recurring transactions

kalkins opened the pull request:
Fixes #47

pep8speaks commented:
Hello @kalkins! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

















There are currently no PEP 8 issues detected in this Pull Request. Cheers! :beers: 

##### Comment last updated at 2019-08-05 07:49:24 UTC

kalkins committed:
Change RecurringTransaction day and month delta to an interval and flag field

kalkins committed:
Allow a changing date of a recurring transaction transaction

The recurring_date of the Transaction object should be
the date that the transaction should be on, according to
the recurring transaction.

kalkins committed:
Fix style

nicomni reviewed (commented):
This is only a namespace change, right?

kalkins commented:
@NicolajNiels1 Partly, it changes the `day_delta` and `month_delta` fields on recurring transactions to `ìnterval` and `interval_type`, and allows instances of recurring transactions on different dates than set by those fields

henriean committed:
Merge branch 'master' into fix/recurring-interval

FredrikAugust commented:
@kalkins I tried to look at the merge conflict, but I'm afraid I'll break it. Think you could take a look?

And please ping me when you're done, so I can review it ASAP. I really want this in master ;)

FredrikAugust committed:
Merge branch 'master' into fix/recurring-interval

FredrikAugust commented on transaction/models.py:
```diff
@@ -68,27 +80,30 @@ class RecurringTransaction(models.Model):
-            interval = self.day_delta
 
-        occurences = rrule(freq, interval=interval, dtstart=self.start_date, until=end_date)
+        occurences = rrule(freq, interval=self.interval, dtstart=self.start_date, until=end_date)
```
Could perhaps replace `freq` with `self.interval_type`?
